### ISSUE TITLE
Use the responder to send errors

### ISSUE THREAD

**Issue by KeesCBakker** [stephenyeargin/hubot-grafana#193]
**Is your feature request related to a problem? Please describe.**
When using a custom responder, it can not trap the sending of the error, resulting in 2 messages:

![image](https://github.com/user-attachments/assets/2b6faba5-a117-401a-8ca3-cfdaa5b991d7)

**Describe the solution you'd like**
The bot currently sends the error using Hubot. When we transfer this functionality to the `Responder` (and let is send it), we are able to override the behavior of sending the error message. 

**Describe alternatives you've considered**
We might be able to create the same feature with response middleware in Hubot, but as we have a `Responder`, it is easier to implement the feature.

**Additional context**
I think I've already got some working code, so I'll be sending a PR shortly.

**Comment by KeesCBakker** [comment]
With the new PR it works like a charm:

![image](https://github.com/user-attachments/assets/f0c0818f-25f4-448c-aa31-f5e7fab06284)

Example of an implementation:

```ts

class WehkampDashboardResponder extends Responder {
  send(res: Hubot.Response, title: string, imageUrl: string, dashboardLink: string) {
    imageUrl = this.fixUrl(imageUrl)
    dashboardLink = this.fixUrl(dashboardLink)

    let msg: any = {
      attachments: [
        {
          fallback: `${title}: ${imageUrl} - ${dashboardLink}`,
          title,
          title_link: dashboardLink,
          image_url: imageUrl,
        },
      ],
      unfurl_links: false,
    }

    this.sendToSlack(res, msg)
  }

  private fixUrl(url: string) {
    url = url.replace("api.grafana.my.domain", "dashboards.my.domain")
    url = url.replace("&fullscreen", "")
    url = replaceTimeExpressions(url)
    url = url.replace(/panelId=(\d+)/, "panelId=$1&viewPanel=$1")
    return url
  }

  sendError(res: Hubot.Response, error: string) {
    if (error == "Dashboard not found") {
      error = ":warning: Dashboard not found!"
    }

    let msg = {
      blocks: [
        {
          type: "section",
          text: {
            type: "plain_text",
            text: error,
            emoji: true,
          },
        },
      ],
      text: error,
    }

    this.sendToSlack(res, msg)
  }

  private sendToSlack(res: Hubot.Response, msg: any) {
    const useThreads = process.env.HUBOT_GRAFANA_USE_THREADS || false
    const message = <any>res.message

    if (useThreads) {
      msg.thread_ts = message.rawMessage.ts
    }

    if (!message.msg || message.msg.used == 1) {
      res.send(msg)
      return
    }

    message.msg.used = 1
    message.msg.send(msg)
  }
}
```
